Remove fields from count() queries when associations are contained.

markstory opened the pull request:
Strip out any fields defined in contained associations as fetching fields and throwing away the results is wasteful and can lead to slow downs in larger datasets.

Refs #5955

markstory committed:
Remove fields from count() queries when associations are contained.

Strip out any fields defined in contained associations as fetching
fields and throwing away the results is wasteful and can lead to slow
downs in larger datasets.

Refs #5955

lorenzo commented on src/ORM/Query.php:
```diff
@@ -507,8 +507,16 @@ public function count()
+                unset($contain['fields']);
+                $cleanContain[$alias] = $contain;
+            }
+            $query->contain(null, true);
```
What about nested associations?

markstory replied:
Darn, I didn't consider those.

lorenzo commented:
I think this could be solved by having a flag in the eager loader to prevent loading fields from associations

markstory commented:
@lorenzo Something like `$query->eagerLoader()->autoFields(false)`? Or did you have something else in mind?

lorenzo commented:
@markstory Yeah, I had something like that in mind

markstory commented:
@lorenzo Ok, I can give that a shot tonight.

markstory committed:
Exclude features from count queries with eagerloader features.

Instead of rewriting contain() options we should use eagerloader
features to exclude associated fields from count queries.

markstory commented:
@lorenzo Updated.

lorenzo commented:
Great job! :clap:

lorenzo committed:
Not returning false in function that is meant to return void
